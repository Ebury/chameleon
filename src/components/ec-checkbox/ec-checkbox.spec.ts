import { type ComponentMountingOptions, mount } from '@vue/test-utils';
import { vi } from 'vitest';
import { defineComponent, ref } from 'vue';

import type { CVueWrapper } from '../../../tests/utils/global';
import EcCheckbox from './ec-checkbox.vue';
import type { CheckboxProps } from './types';

describe('EcCheckbox', () => {
  function mountCheckbox(props?: CheckboxProps, mountOpts?: ComponentMountingOptions<CheckboxProps>) {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    return mount(EcCheckbox as any, {
      props,
      ...mountOpts,
    }) as CVueWrapper;
  }

  function mountCheckboxAsTemplate(
    template: string,
    props: CheckboxProps,
    wrapperComponentOpts?: Record<string, unknown>,
    mountOpts?: ComponentMountingOptions<CheckboxProps>,
  ) {
    const Component = defineComponent({
      components: { EcCheckbox },
      template,
      ...wrapperComponentOpts,
    });

    return mount(Component, {
      props,
      ...mountOpts,
    }) as CVueWrapper;
  }

  describe(':props', () => {
    it(':label - should render the checkbox with a label', () => {
      const wrapper = mountCheckbox({
        label: 'Test label prop',
      });

      expect(wrapper.findByDataTest('ec-checkbox__label').exists()).toBe(true);
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':isSingleLine - should render the checkbox with in a single line with the label as a title', () => {
      const wrapper = mountCheckbox({
        label: 'Test is single line prop',
        isSingleLine: true,
      });

      expect(wrapper.element).toMatchSnapshot();
    });

    it(':error-message - should render the checkbox with an error message', () => {
      const wrapper = mountCheckbox({
        errorMessage: 'Test error message prop',
      });

      expect(wrapper.findByDataTest('ec-checkbox__error-text').exists()).toBe(true);
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':disabled - should render the checkbox with a disabled attribute and not checked', () => {
      const wrapper = mountCheckbox({
        disabled: true,
        modelValue: false,
      });

      expect(wrapper.element).toMatchSnapshot();
    });

    it(':disabled - should render the checkbox with a disabled attribute and indeterminate', () => {
      const wrapper = mountCheckbox({
        disabled: true,
        indeterminate: true,
      });

      expect(wrapper.element).toMatchSnapshot();
    });

    it(':disabled - should render the checkbox with a disabled attribute and checked', () => {
      const wrapper = mountCheckbox({
        disabled: true,
        modelValue: true,
      });

      expect(wrapper.element).toMatchSnapshot();
    });

    it(':indeterminate - should render the checkbox as indeterminate', () => {
      const wrapper = mountCheckbox({
        indeterminate: true,
      });

      expect(wrapper.element).toMatchSnapshot();
    });

    it(':indeterminate - should render even if the checkbox is checked', () => {
      const wrapper = mountCheckbox({
        modelValue: true,
        indeterminate: true,
      });

      expect(wrapper.element).toMatchSnapshot();
    });

    it(':indeterminate - should render the checkbox as checked when indeterminate has switched to false', async () => {
      const wrapper = mountCheckbox({
        modelValue: true,
        indeterminate: true,
      });

      expect(wrapper.element).toMatchSnapshot('before');

      await wrapper.setProps({
        indeterminate: false,
      });

      expect(wrapper.element).toMatchSnapshot('after');
    });
  });

  describe('#slots', () => {
    it('#errorMessage - should render the error message slot if passed', () => {
      const wrapper = mountCheckbox(
        {},
        {
          slots: {
            'error-message': 'Test error message slot',
          },
        },
      );

      expect(wrapper.findByDataTest('ec-checkbox__error-text').exists()).toBe(true);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('#error-message - should render error-message slot instead of error-message prop when both are passed', () => {
      const wrapper = mountCheckbox(
        {
          errorMessage: 'Error message coming from props',
        },
        {
          slots: {
            'error-message': 'Error message coming from slot',
          },
        },
      );

      expect(wrapper.findByDataTest('ec-checkbox__error-text').exists()).toBe(true);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('#label - should render the label slot if passed', () => {
      const wrapper = mountCheckbox(
        {},
        {
          slots: {
            label: 'Test label message slot',
          },
        },
      );

      expect(wrapper.findByDataTest('ec-checkbox__label').exists()).toBe(true);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('#label - should render label slot instead of label prop when both are passed', () => {
      const wrapper = mountCheckbox(
        {
          label: 'Label message coming from props',
        },
        {
          slots: {
            label: 'Label message coming from slot',
          },
        },
      );

      expect(wrapper.findByDataTest('ec-checkbox__label').exists()).toBe(true);
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('@events', () => {
    it('@update:modelValue - should be emitted when input is clicked', () => {
      const wrapper = mountCheckbox();

      wrapper.findByDataTest('ec-checkbox__input').setValue(true);
      expect(wrapper.emitted('update:modelValue')?.length).toBe(1);
    });

    it('@change - should be emitted when input is clicked', () => {
      const changeSpy = vi.fn();
      const wrapper = mountCheckbox({}, {
        attrs: {
          onChange: changeSpy,
        },
      });

      wrapper.findByDataTest('ec-checkbox__input').setValue(true);
      expect(changeSpy).toHaveBeenCalledTimes(1);
    });
  });

  describe('attrs', () => {
    it('should pass all non-prop attributes to the hidden checkbox input', () => {
      const wrapper = mountCheckbox(
        {},
        {
          attrs: {
            'aria-label': 'Random label',
          },
        },
      );
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should render with custom attributes', () => {
      const wrapper = mountCheckbox({}, {
        attrs: {
          'data-test': 'my-data-test',
          class: 'my-class',
          id: 'test-id',
          style: 'top: 0px',
          role: 'checkbox',
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should pass listeners to checkbox input', async () => {
      const clickSpy = vi.fn();
      const wrapper = mountCheckbox({}, {
        attrs: { onClick: clickSpy },
      });

      await wrapper.findByDataTest('ec-checkbox__input').trigger('click');
      expect(clickSpy).toHaveBeenCalledTimes(1);
    });
  });

  describe('v-model', () => {
    it('should render the checkbox and toggle v-model value when input is clicked', async () => {
      const wrapper = mountCheckboxAsTemplate(
        '<ec-checkbox v-model="checked"></ec-checkbox>',
        {},
        {
          setup() {
            const checked = ref(true);
            return { checked };
          },
        },
      );

      expect(wrapper.findByDataTest('ec-checkbox').exists()).toBe(true);
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      expect(wrapper.vm.checked).toBe(true);
      await wrapper.findByDataTest('ec-checkbox__input').setValue(false);
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      expect(wrapper.vm.checked).toBe(false);
    });

    it('should not change the value of v-model if disabled is enabled and user clicks the checkbox', async () => {
      const wrapper = mountCheckboxAsTemplate(
        '<ec-checkbox v-model="checked" disabled></ec-checkbox>',
        {},
        {
          setup() {
            const checked = ref(true);
            return { checked };
          },
        },
      );
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      expect(wrapper.vm.checked).toBe(true);
      await wrapper.findByDataTest('ec-checkbox__input').setValue(false);
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      expect(wrapper.vm.checked).toBe(true);
    });

    it('should have the icon clickable', async () => {
      const element = document.createElement('div');
      document.body.appendChild(element);
      const wrapper = mountCheckboxAsTemplate(
        '<ec-checkbox v-model="model" />',
        {},
        {
          data() {
            return { model: false };
          },
        },
        {
          attachTo: element,
        },
      );

      await wrapper.findByDataTest('ec-checkbox__check-icon-wrapper').trigger('click');
      expect(wrapper.vm.model).toBe(true);

      await wrapper.findByDataTest('ec-checkbox__check-icon').trigger('click');
      expect(wrapper.vm.model).toBe(false);

      wrapper.unmount();
    });

    it('should render the input focused when focused', async () => {
      'ec-checkbox__input';

      const wrapper = mountCheckbox();

      await wrapper.findByDataTest('ec-checkbox__input').trigger('focus');
      expect(wrapper.element).toMatchSnapshot('focused');

      await wrapper.findByDataTest('ec-checkbox__input').trigger('blur');
      expect(wrapper.element).toMatchSnapshot('blurred');
    });
  });
});
