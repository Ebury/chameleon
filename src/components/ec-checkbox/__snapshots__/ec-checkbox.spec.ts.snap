// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EcCheckbox > #slots > #error-message - should render error-message slot instead of error-message prop when both are passed 1`] = `
<div
  class="ec-checkbox"
  data-test="ec-checkbox"
>
  <input
    aria-describedby="ec-checkbox-error-1"
    class="ec-checkbox__input"
    data-test="ec-checkbox__input"
    id="ec-checkbox-1"
    type="checkbox"
  />
  <div
    class="ec-checkbox__label-checkbox-wrapper"
  >
    <span
      class="ec-checkbox__check-icon-wrapper ec-checkbox__check-icon-wrapper--error"
      data-test="ec-checkbox__check-icon-wrapper"
    >
      <!--v-if-->
    </span>
    <label
      class="ec-checkbox__label"
      data-test="ec-checkbox__label"
      for="ec-checkbox-1"
    >
      
      
      
    </label>
  </div>
  <div
    class="ec-checkbox__error-text"
    data-test="ec-checkbox__error-text"
    id="ec-checkbox-error-1"
  >
    
    Error message coming from slot
    
  </div>
</div>
`;

exports[`EcCheckbox > #slots > #errorMessage - should render the error message slot if passed 1`] = `
<div
  class="ec-checkbox"
  data-test="ec-checkbox"
>
  <input
    aria-describedby="ec-checkbox-error-1"
    class="ec-checkbox__input"
    data-test="ec-checkbox__input"
    id="ec-checkbox-1"
    type="checkbox"
  />
  <div
    class="ec-checkbox__label-checkbox-wrapper"
  >
    <span
      class="ec-checkbox__check-icon-wrapper ec-checkbox__check-icon-wrapper--error"
      data-test="ec-checkbox__check-icon-wrapper"
    >
      <!--v-if-->
    </span>
    <label
      class="ec-checkbox__label"
      data-test="ec-checkbox__label"
      for="ec-checkbox-1"
    >
      
      
      
    </label>
  </div>
  <div
    class="ec-checkbox__error-text"
    data-test="ec-checkbox__error-text"
    id="ec-checkbox-error-1"
  >
    
    Test error message slot
    
  </div>
</div>
`;

exports[`EcCheckbox > #slots > #label - should render label slot instead of label prop when both are passed 1`] = `
<div
  class="ec-checkbox"
  data-test="ec-checkbox"
>
  <input
    class="ec-checkbox__input"
    data-test="ec-checkbox__input"
    id="ec-checkbox-1"
    type="checkbox"
  />
  <div
    class="ec-checkbox__label-checkbox-wrapper"
  >
    <span
      class="ec-checkbox__check-icon-wrapper"
      data-test="ec-checkbox__check-icon-wrapper"
    >
      <!--v-if-->
    </span>
    <label
      class="ec-checkbox__label"
      data-test="ec-checkbox__label"
      for="ec-checkbox-1"
    >
      
      Label message coming from slot
      
    </label>
  </div>
  <!--v-if-->
</div>
`;

exports[`EcCheckbox > #slots > #label - should render the label slot if passed 1`] = `
<div
  class="ec-checkbox"
  data-test="ec-checkbox"
>
  <input
    class="ec-checkbox__input"
    data-test="ec-checkbox__input"
    id="ec-checkbox-1"
    type="checkbox"
  />
  <div
    class="ec-checkbox__label-checkbox-wrapper"
  >
    <span
      class="ec-checkbox__check-icon-wrapper"
      data-test="ec-checkbox__check-icon-wrapper"
    >
      <!--v-if-->
    </span>
    <label
      class="ec-checkbox__label"
      data-test="ec-checkbox__label"
      for="ec-checkbox-1"
    >
      
      Test label message slot
      
    </label>
  </div>
  <!--v-if-->
</div>
`;

exports[`EcCheckbox > :props > :disabled - should render the checkbox with a disabled attribute and checked 1`] = `
<div
  class="ec-checkbox"
  data-test="ec-checkbox"
>
  <input
    class="ec-checkbox__input"
    data-test="ec-checkbox__input"
    disabled=""
    id="ec-checkbox-1"
    type="checkbox"
  />
  <div
    class="ec-checkbox__label-checkbox-wrapper"
  >
    <span
      class="ec-checkbox__check-icon-wrapper ec-checkbox__check-icon-wrapper--checked ec-checkbox__check-icon-wrapper--disabled ec-checkbox__check-icon-wrapper--checked-and-disabled"
      data-test="ec-checkbox__check-icon-wrapper"
    >
      <svg
        class="ec-icon ec-checkbox__check-icon"
        data-test="ec-checkbox__check-icon"
        height="16"
        width="16"
      >
        <use
          xlink:href="#ec-simple-check"
        />
      </svg>
    </span>
    <label
      class="ec-checkbox__label"
      data-test="ec-checkbox__label"
      for="ec-checkbox-1"
    >
      
      
      
    </label>
  </div>
  <!--v-if-->
</div>
`;

exports[`EcCheckbox > :props > :disabled - should render the checkbox with a disabled attribute and indeterminate 1`] = `
<div
  class="ec-checkbox"
  data-test="ec-checkbox"
>
  <input
    class="ec-checkbox__input"
    data-test="ec-checkbox__input"
    disabled=""
    id="ec-checkbox-1"
    type="checkbox"
  />
  <div
    class="ec-checkbox__label-checkbox-wrapper"
  >
    <span
      class="ec-checkbox__check-icon-wrapper ec-checkbox__check-icon-wrapper--indeterminate ec-checkbox__check-icon-wrapper--disabled ec-checkbox__check-icon-wrapper--indeterminate-and-disabled"
      data-test="ec-checkbox__check-icon-wrapper"
    >
      <span
        class="ec-checkbox__indeterminate-icon"
        data-test="ec-checkbox__indeterminate-icon"
      />
    </span>
    <label
      class="ec-checkbox__label"
      data-test="ec-checkbox__label"
      for="ec-checkbox-1"
    >
      
      
      
    </label>
  </div>
  <!--v-if-->
</div>
`;

exports[`EcCheckbox > :props > :disabled - should render the checkbox with a disabled attribute and not checked 1`] = `
<div
  class="ec-checkbox"
  data-test="ec-checkbox"
>
  <input
    class="ec-checkbox__input"
    data-test="ec-checkbox__input"
    disabled=""
    id="ec-checkbox-1"
    type="checkbox"
  />
  <div
    class="ec-checkbox__label-checkbox-wrapper"
  >
    <span
      class="ec-checkbox__check-icon-wrapper ec-checkbox__check-icon-wrapper--disabled"
      data-test="ec-checkbox__check-icon-wrapper"
    >
      <!--v-if-->
    </span>
    <label
      class="ec-checkbox__label"
      data-test="ec-checkbox__label"
      for="ec-checkbox-1"
    >
      
      
      
    </label>
  </div>
  <!--v-if-->
</div>
`;

exports[`EcCheckbox > :props > :error-message - should render the checkbox with an error message 1`] = `
<div
  class="ec-checkbox"
  data-test="ec-checkbox"
>
  <input
    aria-describedby="ec-checkbox-error-1"
    class="ec-checkbox__input"
    data-test="ec-checkbox__input"
    id="ec-checkbox-1"
    type="checkbox"
  />
  <div
    class="ec-checkbox__label-checkbox-wrapper"
  >
    <span
      class="ec-checkbox__check-icon-wrapper ec-checkbox__check-icon-wrapper--error"
      data-test="ec-checkbox__check-icon-wrapper"
    >
      <!--v-if-->
    </span>
    <label
      class="ec-checkbox__label"
      data-test="ec-checkbox__label"
      for="ec-checkbox-1"
    >
      
      
      
    </label>
  </div>
  <div
    class="ec-checkbox__error-text"
    data-test="ec-checkbox__error-text"
    id="ec-checkbox-error-1"
  >
    
    Test error message prop
    
  </div>
</div>
`;

exports[`EcCheckbox > :props > :indeterminate - should render even if the checkbox is checked 1`] = `
<div
  class="ec-checkbox"
  data-test="ec-checkbox"
>
  <input
    class="ec-checkbox__input"
    data-test="ec-checkbox__input"
    id="ec-checkbox-1"
    type="checkbox"
  />
  <div
    class="ec-checkbox__label-checkbox-wrapper"
  >
    <span
      class="ec-checkbox__check-icon-wrapper ec-checkbox__check-icon-wrapper--checked ec-checkbox__check-icon-wrapper--indeterminate"
      data-test="ec-checkbox__check-icon-wrapper"
    >
      <span
        class="ec-checkbox__indeterminate-icon"
        data-test="ec-checkbox__indeterminate-icon"
      />
    </span>
    <label
      class="ec-checkbox__label"
      data-test="ec-checkbox__label"
      for="ec-checkbox-1"
    >
      
      
      
    </label>
  </div>
  <!--v-if-->
</div>
`;

exports[`EcCheckbox > :props > :indeterminate - should render the checkbox as checked when indeterminate has switched to false > after 1`] = `
<div
  class="ec-checkbox"
  data-test="ec-checkbox"
>
  <input
    class="ec-checkbox__input"
    data-test="ec-checkbox__input"
    id="ec-checkbox-1"
    type="checkbox"
  />
  <div
    class="ec-checkbox__label-checkbox-wrapper"
  >
    <span
      class="ec-checkbox__check-icon-wrapper ec-checkbox__check-icon-wrapper--checked"
      data-test="ec-checkbox__check-icon-wrapper"
    >
      <svg
        class="ec-icon ec-checkbox__check-icon"
        data-test="ec-checkbox__check-icon"
        height="16"
        width="16"
      >
        <use
          xlink:href="#ec-simple-check"
        />
      </svg>
    </span>
    <label
      class="ec-checkbox__label"
      data-test="ec-checkbox__label"
      for="ec-checkbox-1"
    >
      
      
      
    </label>
  </div>
  <!--v-if-->
</div>
`;

exports[`EcCheckbox > :props > :indeterminate - should render the checkbox as checked when indeterminate has switched to false > before 1`] = `
<div
  class="ec-checkbox"
  data-test="ec-checkbox"
>
  <input
    class="ec-checkbox__input"
    data-test="ec-checkbox__input"
    id="ec-checkbox-1"
    type="checkbox"
  />
  <div
    class="ec-checkbox__label-checkbox-wrapper"
  >
    <span
      class="ec-checkbox__check-icon-wrapper ec-checkbox__check-icon-wrapper--checked ec-checkbox__check-icon-wrapper--indeterminate"
      data-test="ec-checkbox__check-icon-wrapper"
    >
      <span
        class="ec-checkbox__indeterminate-icon"
        data-test="ec-checkbox__indeterminate-icon"
      />
    </span>
    <label
      class="ec-checkbox__label"
      data-test="ec-checkbox__label"
      for="ec-checkbox-1"
    >
      
      
      
    </label>
  </div>
  <!--v-if-->
</div>
`;

exports[`EcCheckbox > :props > :indeterminate - should render the checkbox as indeterminate 1`] = `
<div
  class="ec-checkbox"
  data-test="ec-checkbox"
>
  <input
    class="ec-checkbox__input"
    data-test="ec-checkbox__input"
    id="ec-checkbox-1"
    type="checkbox"
  />
  <div
    class="ec-checkbox__label-checkbox-wrapper"
  >
    <span
      class="ec-checkbox__check-icon-wrapper ec-checkbox__check-icon-wrapper--indeterminate"
      data-test="ec-checkbox__check-icon-wrapper"
    >
      <span
        class="ec-checkbox__indeterminate-icon"
        data-test="ec-checkbox__indeterminate-icon"
      />
    </span>
    <label
      class="ec-checkbox__label"
      data-test="ec-checkbox__label"
      for="ec-checkbox-1"
    >
      
      
      
    </label>
  </div>
  <!--v-if-->
</div>
`;

exports[`EcCheckbox > :props > :isSingleLine - should render the checkbox with in a single line with the label as a title 1`] = `
<div
  class="ec-checkbox"
  data-test="ec-checkbox"
>
  <input
    class="ec-checkbox__input"
    data-test="ec-checkbox__input"
    id="ec-checkbox-1"
    type="checkbox"
  />
  <div
    class="ec-checkbox__label-checkbox-wrapper ec-checkbox__label-checkbox-wrapper--is-single-line"
  >
    <span
      class="ec-checkbox__check-icon-wrapper"
      data-test="ec-checkbox__check-icon-wrapper"
    >
      <!--v-if-->
    </span>
    <label
      class="ec-checkbox__label ec-checkbox__label--is-single-line"
      data-test="ec-checkbox__label"
      for="ec-checkbox-1"
      title="Test is single line prop"
    >
      
      Test is single line prop
      
    </label>
  </div>
  <!--v-if-->
</div>
`;

exports[`EcCheckbox > :props > :label - should render the checkbox with a label 1`] = `
<div
  class="ec-checkbox"
  data-test="ec-checkbox"
>
  <input
    class="ec-checkbox__input"
    data-test="ec-checkbox__input"
    id="ec-checkbox-1"
    type="checkbox"
  />
  <div
    class="ec-checkbox__label-checkbox-wrapper"
  >
    <span
      class="ec-checkbox__check-icon-wrapper"
      data-test="ec-checkbox__check-icon-wrapper"
    >
      <!--v-if-->
    </span>
    <label
      class="ec-checkbox__label"
      data-test="ec-checkbox__label"
      for="ec-checkbox-1"
    >
      
      Test label prop
      
    </label>
  </div>
  <!--v-if-->
</div>
`;

exports[`EcCheckbox > attrs > should pass all non-prop attributes to the hidden checkbox input 1`] = `
<div
  class="ec-checkbox"
  data-test="ec-checkbox"
>
  <input
    aria-label="Random label"
    class="ec-checkbox__input"
    data-test="ec-checkbox__input"
    id="ec-checkbox-1"
    type="checkbox"
  />
  <div
    class="ec-checkbox__label-checkbox-wrapper"
  >
    <span
      class="ec-checkbox__check-icon-wrapper"
      data-test="ec-checkbox__check-icon-wrapper"
    >
      <!--v-if-->
    </span>
    <label
      class="ec-checkbox__label"
      data-test="ec-checkbox__label"
      for="ec-checkbox-1"
    >
      
      
      
    </label>
  </div>
  <!--v-if-->
</div>
`;

exports[`EcCheckbox > attrs > should render with custom attributes 1`] = `
<div
  class="ec-checkbox my-class"
  data-test="my-data-test ec-checkbox"
  style="top: 0px;"
>
  <input
    class="ec-checkbox__input"
    data-test="ec-checkbox__input"
    id="ec-checkbox-1"
    role="checkbox"
    type="checkbox"
  />
  <div
    class="ec-checkbox__label-checkbox-wrapper"
  >
    <span
      class="ec-checkbox__check-icon-wrapper"
      data-test="ec-checkbox__check-icon-wrapper"
    >
      <!--v-if-->
    </span>
    <label
      class="ec-checkbox__label"
      data-test="ec-checkbox__label"
      for="ec-checkbox-1"
    >
      
      
      
    </label>
  </div>
  <!--v-if-->
</div>
`;

exports[`EcCheckbox > v-model > should render the input focused when focused > blurred 1`] = `
<div
  class="ec-checkbox"
  data-test="ec-checkbox"
>
  <input
    class="ec-checkbox__input"
    data-test="ec-checkbox__input"
    id="ec-checkbox-1"
    type="checkbox"
  />
  <div
    class="ec-checkbox__label-checkbox-wrapper"
  >
    <span
      class="ec-checkbox__check-icon-wrapper"
      data-test="ec-checkbox__check-icon-wrapper"
    >
      <!--v-if-->
    </span>
    <label
      class="ec-checkbox__label"
      data-test="ec-checkbox__label"
      for="ec-checkbox-1"
    >
      
      
      
    </label>
  </div>
  <!--v-if-->
</div>
`;

exports[`EcCheckbox > v-model > should render the input focused when focused > focused 1`] = `
<div
  class="ec-checkbox"
  data-test="ec-checkbox"
>
  <input
    class="ec-checkbox__input"
    data-test="ec-checkbox__input"
    id="ec-checkbox-1"
    type="checkbox"
  />
  <div
    class="ec-checkbox__label-checkbox-wrapper"
  >
    <span
      class="ec-checkbox__check-icon-wrapper ec-checkbox__check-icon-wrapper--focused"
      data-test="ec-checkbox__check-icon-wrapper"
    >
      <!--v-if-->
    </span>
    <label
      class="ec-checkbox__label"
      data-test="ec-checkbox__label"
      for="ec-checkbox-1"
    >
      
      
      
    </label>
  </div>
  <!--v-if-->
</div>
`;
